Fix failing test with mismatched dtype from array comprehension

On Windows 64-bit, the following test failure occurs (noted in #2479):
```
======================================================================
FAIL: test_array_comp_inferred_dtype_outside_setitem (numba.tests.test_comprehension.TestArrayComprehension)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\conda64\conda-bld\numba_1501601782293\_t_env\lib\site-packages\numba\tests\test_comprehension.py", line 437, in test_array_comp_inferred_dtype_outside_setitem
    self.check(array_comp, 10, v, assert_dtype=np.int)
  File "C:\conda64\conda-bld\numba_1501601782293\_t_env\lib\site-packages\numba\tests\test_comprehension.py", line 249, in check
    self.assertEqual(cres[1].dtype, assert_dtype)
AssertionError: dtype('int64') != <class 'int'>
```
Link to code: https://github.com/numba/numba/blob/master/numba/tests/test_comprehension.py#L249

The issue is that on Win64, Python int is C long which is int32, not int64.  The expected int size should be explicitly int64 on all platforms, I think.

--- a/numba/tests/test_comprehension.py
+++ b/numba/tests/test_comprehension.py
@@ -434,7 +434,7 @@ class TestArrayComprehension(unittest.TestCase):
 
         # float to int cast is valid
         v = 1.2
-        self.check(array_comp, 10, v, assert_dtype=np.int64)
+        self.check(array_comp, 10, v, assert_dtype=np.intp)
         # complex to int cast is invalid
         with self.assertRaises(TypingError) as raises:
             cfunc = jit(nopython=True)(array_comp)

--- a/numba/tests/test_comprehension.py
+++ b/numba/tests/test_comprehension.py
@@ -434,7 +434,7 @@ class TestArrayComprehension(unittest.TestCase):
 
         # float to int cast is valid
         v = 1.2
-        self.check(array_comp, 10, v, assert_dtype=np.int)
+        self.check(array_comp, 10, v, assert_dtype=np.int64)
         # complex to int cast is invalid
         with self.assertRaises(TypingError) as raises:
             cfunc = jit(nopython=True)(array_comp)

